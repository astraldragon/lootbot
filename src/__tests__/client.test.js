jest.mock("fs");
jest.mock("path");
jest.mock("discord-akairo");

jest.mock("../handlers", () => ({
  CommandHandler: jest.fn("CommandHandler"),
  InhibitorHandler: jest.fn("InhibtorHandler")
}));

jest.mock("../models", () => ({
  Settings: jest.fn("Settings")
}));

const { AkairoClient, SequelizeProvider } = require("discord-akairo");
const { CommandHandler, InhibitorHandler } = require("../handlers");
const Client = require("../client");

describe("client", () => {
  beforeEach(() => {
    SequelizeProvider.mockReturnValue({
      init: jest.fn("init")
    });

    this.client = new Client();
    this.client.akairoOptions = { test: true };
  });

  it("should configure properly", () => {
    expect(AkairoClient).toMatchSnapshot();
  });

  it("should register sequelize as settings provider", () => {
    expect(this.client.settings).toBe(SequelizeProvider.mock.instances[0]);
    expect(SequelizeProvider).toMatchSnapshot();
  });

  it("should create custom command handler", () => {
    this.client.build();
    expect(CommandHandler).toMatchSnapshot();
    expect(AkairoClient.prototype.build).toHaveBeenCalled();
  });

  it("should create custom inhibitor handler", () => {
    this.client.build();
    expect(InhibitorHandler).toMatchSnapshot();
    expect(AkairoClient.prototype.build).toHaveBeenCalled();
  });

  it("should log in properly", async () => {
    const token = "1234";
    await this.client.login(token);
    expect(this.client.settings.init).toHaveBeenCalled();
    expect(AkairoClient.prototype.login).toHaveBeenCalledWith(token);
  });
});
