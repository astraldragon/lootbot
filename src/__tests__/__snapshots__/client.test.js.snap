// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`client should configure properly 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "allowMention": true,
        "commandDirectory": "./src/commands",
        "inhibitorDirectory": "./src/inhibitors",
        "listenerDirectory": "./src/listeners",
        "ownerID": undefined,
        "prefix": Array [
          "!lootbot",
          "!lb",
        ],
      },
      Object {
        "disableEveryone": true,
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`client should create custom command handler 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      CustomClient {
        "_eval": [MockFunction],
        "_pong": [MockFunction],
        "_setPresence": [MockFunction],
        "_validateOptions": [MockFunction],
        "addDatabase": [MockFunction],
        "addListener": [MockFunction],
        "akairoOptions": Object {
          "test": true,
        },
        "build": [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        },
        "clearInterval": [MockFunction],
        "clearTimeout": [MockFunction],
        "commandHandler": mockConstructor {},
        "createVoiceBroadcast": [MockFunction],
        "destroy": [MockFunction],
        "emit": [MockFunction],
        "eventNames": [MockFunction],
        "fetchApplication": [MockFunction],
        "fetchInvite": [MockFunction],
        "fetchUser": [MockFunction],
        "fetchVoiceRegions": [MockFunction],
        "fetchWebhook": [MockFunction],
        "generateInvite": [MockFunction],
        "getMaxListeners": [MockFunction],
        "inhibitorHandler": mockConstructor {},
        "listenerCount": [MockFunction],
        "listeners": [MockFunction],
        "loadAll": [MockFunction],
        "login": [MockFunction],
        "once": [MockFunction],
        "prependListener": [MockFunction],
        "prependOnceListener": [MockFunction],
        "rawListeners": [MockFunction],
        "removeAllListeners": [MockFunction],
        "removeListener": [MockFunction],
        "setInterval": [MockFunction],
        "setMaxListeners": [MockFunction],
        "setTimeout": [MockFunction],
        "settings": SequelizeProvider {
          "clear": [MockFunction],
          "delete": [MockFunction],
          "get": [MockFunction],
          "init": [MockFunction],
          "set": [MockFunction],
        },
        "sweepMessages": [MockFunction],
        "syncGuilds": [MockFunction],
      },
      Object {
        "test": true,
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`client should create custom inhibitor handler 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      CustomClient {
        "_eval": [MockFunction],
        "_pong": [MockFunction],
        "_setPresence": [MockFunction],
        "_validateOptions": [MockFunction],
        "addDatabase": [MockFunction],
        "addListener": [MockFunction],
        "akairoOptions": Object {
          "test": true,
        },
        "build": [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        },
        "clearInterval": [MockFunction],
        "clearTimeout": [MockFunction],
        "commandHandler": mockConstructor {},
        "createVoiceBroadcast": [MockFunction],
        "destroy": [MockFunction],
        "emit": [MockFunction],
        "eventNames": [MockFunction],
        "fetchApplication": [MockFunction],
        "fetchInvite": [MockFunction],
        "fetchUser": [MockFunction],
        "fetchVoiceRegions": [MockFunction],
        "fetchWebhook": [MockFunction],
        "generateInvite": [MockFunction],
        "getMaxListeners": [MockFunction],
        "inhibitorHandler": mockConstructor {},
        "listenerCount": [MockFunction],
        "listeners": [MockFunction],
        "loadAll": [MockFunction],
        "login": [MockFunction],
        "once": [MockFunction],
        "prependListener": [MockFunction],
        "prependOnceListener": [MockFunction],
        "rawListeners": [MockFunction],
        "removeAllListeners": [MockFunction],
        "removeListener": [MockFunction],
        "setInterval": [MockFunction],
        "setMaxListeners": [MockFunction],
        "setTimeout": [MockFunction],
        "settings": SequelizeProvider {
          "clear": [MockFunction],
          "delete": [MockFunction],
          "get": [MockFunction],
          "init": [MockFunction],
          "set": [MockFunction],
        },
        "sweepMessages": [MockFunction],
        "syncGuilds": [MockFunction],
      },
      Object {
        "test": true,
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`client should register sequelize as settings provider 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      [MockFunction],
      Object {
        "dataColumn": "settings",
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;
